oyente/input_helper.py

```diff
@@ -108,23 +109,29 @@ def _get_compiled_contracts(self):
         return self.compiled_contracts
 
     def _compile_solidity(self):
-        if not self.allow_paths:
-            cmd = "solc --bin-runtime %s %s" % (self.remap, self.source)
-        else:
-            cmd = "solc --bin-runtime %s %s --allow-paths %s" % (self.remap, self.source, self.allow_paths)
-        err = ''
-        if self.compilation_err:
-            out, err = run_command_with_err(cmd)
-            err = re.sub(self.root_path, "", err)
-        else:
-            out = run_command(cmd)
+        try:
+            com = CryticCompile(self.source)
+            contracts = [(com.target + ':' + name, com.bytecode_runtime(name)) for name in com.contracts_names if com.bytecode_runtime(name)]
+            
+            libs = [lib for _, bytecode in contracts for lib in re.findall(r"_+(.*?)_+", bytecode) if lib]
+            if libs:
+                libs = set(libs)
+                return self._link_libraries(self.source, libs)
```

Comment:
This will still call `solc` directly though.

Reply:
Couldn't figure out a way to link libraries using Crytic Compile that's why i left it as is.

Reply:
How about this: Do `CryticCompile` first, then realized that there are unlinked libraries (this branching condition will be true). Then, do `CryticCompile` for second time with `solc_args` set to the `--libraries` option constructed in the `_link_libraries` function.

@Mayur1496 Let me know what you think. If it sounds good, maybe can modify `_link_libraries` to use `CryticCompile` (instead of inlining here)